File: UniEatsBackEnd/Models/OrderItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class OrderItem
    {
        public int OrderItemId { get; set; }

        [Required]
        public int OrderId { get; set; }

        [Required]
        public int ItemId { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Quantity must be greater than 0.")]
        public int Quantity { get; set; }

        [Range(0.01, double.MaxValue, ErrorMessage = "Price must be greater than 0.")]
        public decimal Price { get; set; }

        public decimal Subtotal => Quantity * Price;

        public string ItemDescription { get; set; }

        // Navigation properties
        public Order Order { get; set; }
        public FoodItem FoodItem { get; set; }
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/orderitem.ts
import { FoodItem } from "./food-item";
import { Order } from "./order";

export interface OrderItem {
  orderItemId: number; // Unique identifier for the order item
  orderId: number; // ID of the associated order
  itemId: number; // ID of the associated food item
  quantity: number; // Quantity of the item, must be greater than 0
  price: number; // Price of a single item, must be > 0
  subtotal?: number; // Computed property: quantity * price
  itemDescription?: string; // Optional description of the item
  order?: Order | null; // Optional navigation property for the associated order
  foodItem?: FoodItem | null; // Optional navigation property for the associated food item
}
